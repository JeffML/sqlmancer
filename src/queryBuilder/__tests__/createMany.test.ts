import { withDialects } from './__utilties__'
import { ActorCreateManyBuilder } from './__fixtures__/models'

describe('CreateManyBuilder', () => {
  withDialects((options, rollback) => {
    describe('basic queries', () => {
      test('no additional options', async () => {
        const builder = new ActorCreateManyBuilder(options, [
          { firstName: 'SUSAN', lastName: 'ANTHONY' },
          { firstName: 'VIRGINIA', lastName: 'WOLF' },
        ])
        const { sql, bindings } = builder.toQueryBuilder().toSQL()
        await rollback(builder, result => expect(result).toBeArray())
        expect(sql).toMatchSnapshot()
        expect(bindings).toMatchSnapshot()
      })

      test('non-existent field', async () => {
        const builder = new ActorCreateManyBuilder(options, [
          { firstName: 'SUSAN', lastName: 'ANTHONY', foo: 'bar' },
          { firstName: 'VIRGINIA', lastName: 'WOLF', foo: 'bar' },
        ] as any)
        const { sql, bindings } = builder.toQueryBuilder().toSQL()
        await rollback(builder, result => expect(result).toBeArray())
        expect(sql).toMatchSnapshot()
        expect(bindings).toMatchSnapshot()
      })
    })
  })
})
