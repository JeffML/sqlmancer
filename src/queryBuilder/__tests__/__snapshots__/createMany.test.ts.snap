// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateManyBuilder pg basic queries no additional options 1`] = `"insert into \\"actor\\" (\\"first_name\\", \\"last_name\\") values (?, ?), (?, ?) returning \\"actor_id\\""`;

exports[`CreateManyBuilder pg basic queries no additional options 2`] = `
Array [
  "SUSAN",
  "ANTHONY",
  "VIRGINIA",
  "WOLF",
]
`;

exports[`CreateManyBuilder pg basic queries non-existent field 1`] = `"insert into \\"actor\\" (\\"first_name\\", \\"last_name\\") values (?, ?), (?, ?) returning \\"actor_id\\""`;

exports[`CreateManyBuilder pg basic queries non-existent field 2`] = `
Array [
  "SUSAN",
  "ANTHONY",
  "VIRGINIA",
  "WOLF",
]
`;
